core-web/libs/edit-content/src/lib/fields/dot-edit-content-host-folder-field/dot-edit-content-host-folder-field.component.ts

```diff
@@ -0,0 +1,122 @@
+import {
+    ChangeDetectionStrategy,
+    Component,
+    Input,
+    OnInit,
+    ViewChild,
+    inject,
+    signal
+} from '@angular/core';
+import { ControlContainer, FormControl, ReactiveFormsModule } from '@angular/forms';
+
+import { TreeSelect, TreeSelectModule } from 'primeng/treeselect';
+
+import { DotCMSContentTypeField } from '@dotcms/dotcms-models';
+
+import { DotEditContentFieldSingleSelectableDataTypes } from '../../models/dot-edit-content-field.type';
+import {
+    TreeNodeItem,
+    TreeNodeSelectItem
+} from '../../models/dot-edit-content-host-folder-field.interface';
+import { TruncatePathPipe } from '../../pipes/truncate-path.pipe';
+import { DotEditContentService } from '../../services/dot-edit-content.service';
+import { createPaths } from '../../utils/functions.util';
+
+@Component({
+    selector: 'dot-edit-content-host-folder-field',
+    standalone: true,
+    imports: [TreeSelectModule, ReactiveFormsModule, TruncatePathPipe],
+    templateUrl: './dot-edit-content-host-folder-field.component.html',
+    styleUrls: ['./dot-edit-content-host-folder-field.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    viewProviders: [
+        {
+            provide: ControlContainer,
+            useFactory: () => inject(ControlContainer, { skipSelf: true })
+        }
+    ]
+})
+export class DotEditContentHostFolderFieldComponent implements OnInit {
+    @Input() field!: DotCMSContentTypeField;
+    @ViewChild(TreeSelect) treeSelect!: TreeSelect;
+    readonly #controlContainer = inject(ControlContainer);
+    readonly #editContentService = inject(DotEditContentService);
```

Comment:
TIL, what's the `#` for?

Reply:
@fmontes The native way to support private attrs in ES6, TS with private only support at build time, but with that these variables are private in JS.

<img width="600" alt="Screenshot 2024-06-07 at 10 32 36 AM" src="https://github.com/dotCMS/core/assets/7611944/8ba36db7-695a-4319-9b2a-c0d4f8fdf93b">